Dockerfile

```diff
@@ -391,24 +383,28 @@ RUN \
 # module so `rustc` knows they exist.
 
 RUN \
-  for libc in gnu musl ; do \
-    cp compiler/rustc_target/src/spec/targets/${ARCH}_{unknown,${VENDOR}}_linux_${libc}.rs && \
-    sed -i -e '/let mut base = base::linux_'${libc}'::opts();/a base.vendor = "'${VENDOR}'".into();' \
-      compiler/rustc_target/src/spec/targets/${ARCH}_${VENDOR}_linux_${libc}.rs && \
-    sed -i -e '/ \.\.base::linux_'${libc}'::opts()/i vendor: "'${VENDOR}'".into(),' \
-      compiler/rustc_target/src/spec/targets/${ARCH}_${VENDOR}_linux_${libc}.rs && \
-    sed -i -e '/("'${ARCH}-unknown-linux-${libc}'", .*),/a("'${ARCH}-${VENDOR}-linux-${libc}'", '${ARCH}_${VENDOR}_linux_${libc}'),' \
-      compiler/rustc_target/src/spec/mod.rs ; \
+  for arch in x86_64 aarch64 ; do \
+    for libc in gnu musl ; do \
+      cp compiler/rustc_target/src/spec/targets/${arch}_{unknown,${VENDOR}}_linux_${libc}.rs && \
+      sed -i -e '/let mut base = base::linux_'${libc}'::opts();/a base.vendor = "'${VENDOR}'".into();' \
+        compiler/rustc_target/src/spec/targets/${arch}_${VENDOR}_linux_${libc}.rs && \
+      sed -i -e '/ \.\.base::linux_'${libc}'::opts()/i vendor: "'${VENDOR}'".into(),' \
+        compiler/rustc_target/src/spec/targets/${arch}_${VENDOR}_linux_${libc}.rs && \
+      sed -i -e '/("'${arch}-unknown-linux-${libc}'", .*),/a("'${arch}-${VENDOR}-linux-${libc}'", '${arch}_${VENDOR}_linux_${libc}'),' \
```

Comment:
> Are these effectively patches to support a "vendor" of Bottlerocket? Is declaring oneself a vendor not a supported build argument?

Yes - per the comment above, we need `*-bottlerocket-linux-*` targets to distinguish from `*-unknown-linux-*` targets when compiling for a target that matches the host architecture. The "vendor" is part of the target triple; you can override target on the command line but not the individual components like vendor or os.

It's possible to create a [custom target](https://docs.rust-embedded.org/embedonomicon/custom-target.html) using nightly cargo. Even when that stabilizes, I expect I'll still prefer to do it here in the SDK so that we don't have to build std and core "on demand" when building packages.